### PR TITLE
feat: File blocks

### PR DESCRIPTION
**Pull request by matthewlipski**
This PR adds file blocks which can be extended to show previews for different file types. Currently, this includes an image preview by default. The file block replaces the existing image block.

### CODE REVIEW

File: packages/core/src/editor/BlockNoteEditor.ts [hunk 1]
```diff
@@ -254,9 +254,9 @@ export class BlockNoteEditor<
     this.linkToolbar = new LinkToolbarProsemirrorPlugin(this);
     this.sideMenu = new SideMenuProsemirrorPlugin(this);
     this.suggestionMenus = new SuggestionMenuProseMirrorPlugin(this);
-    if (checkDefaultBlockTypeInSchema("image", this)) {
+    if (checkDefaultBlockTypeInSchema("file", this)) {
```
Comment (YousefED) [inline packages/core/src/editor/BlockNoteEditor.ts:257]: TODO. Maybe most reasonable to always add the plugin?

File: packages/mantine/package.json [hunk 2]
```diff
@@ -55,7 +55,7 @@
     "@mantine/utils": "^6.0.21",
     "react": "^18",
     "react-dom": "^18",
-    "react-icons": "^4.3.1",
+    "react-icons": "^5.2.1",
```
Comment (YousefED) [inline packages/mantine/package.json:58]: todo: update everywhere

File: packages/react/src/blocks/AudioBlockContent/AudioBlockContent.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+};
+
+export const ReactAudioBlock = createReactBlockSpec(audioBlockConfig, {
+  render: (props) => (
+    <div className={"bn-file-block-content-wrapper"}>
+      {props.block.props.url === "" ? (
+        <AddFileButton
+          {...props}
+          editor={props.editor as any}
+          buttonText={"Add audio"}
```
Comment (YousefED) [inline packages/react/src/blocks/AudioBlockContent/AudioBlockContent.tsx:60]: TODO: translate

File: packages/core/src/blocks/AudioBlockContent/audioBlockHelpers.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export const parseAudioElement = (audioElement: HTMLAudioElement) => {
```
Comment (YousefED) [inline packages/core/src/blocks/AudioBlockContent/audioBlockHelpers.ts:1]: shouldn't this check for an "audio" html element?

Comment (YousefED) [reply packages/core/src/blocks/AudioBlockContent/audioBlockHelpers.ts:1]: I think this should have also come up with unit tests?

File: packages/react/src/blocks/FileBlockContent/fileBlockHelpers.tsx [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+  }
+) => {
+  return (
+    <div className={"bn-file-and-caption-wrapper"}>
+      {props.children}
+      {props.block.props.caption && (
+        <p
+          className={"bn-file-caption"}
+          style={{
+            paddingBlock: props.block.props.caption ? "4px" : undefined,
```
Comment (YousefED) [inline packages/react/src/blocks/FileBlockContent/fileBlockHelpers.tsx:36]: - props.block.props.caption is always defined here.
- better to move to stylesheet anyway?

File: packages/react/src/blocks/FileBlockContent/fileBlockHelpers.tsx [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+        handleUsed: "left" | "right";
+      }
+    | undefined
+  >(undefined);
+
+  useEffect(() => {
+    // Updates the child width with an updated width depending on the cursor X
+    // offset from when the resize began, and which resize handle is being used.
+    const windowMouseMoveHandler = (event: MouseEvent) => {
+      if (!resizeParams) {
```
Comment (YousefED) [inline packages/react/src/blocks/FileBlockContent/fileBlockHelpers.tsx:127]: isn't it cleaner to not create the event handler when there is no mouse down?

File: packages/react/src/components/FilePanel/FilePanelController.tsx [hunk 7]
```diff
@@ -11,29 +11,29 @@ import { FC } from "react";
...
+import { FilePanel } from "./FilePanel";
+import { FilePanelProps } from "./FilePanelProps";
 
-export const ImagePanelController = <
+export const FilePanelController = <
   I extends InlineContentSchema = DefaultInlineContentSchema,
   S extends StyleSchema = DefaultStyleSchema
 >(props: {
-  imageToolbar?: FC<ImagePanelProps<I, S>>;
+  fileToolbar?: FC<FilePanelProps<I, S>>;
```
Comment (YousefED) [inline packages/react/src/components/FilePanel/FilePanelController.tsx:21]: should this be named `filePanel` for consistency?

File: packages/react/src/editor/BlockNoteView.tsx [hunk 8]
```diff
@@ -110,7 +110,7 @@ function BlockNoteViewComponent<
           linkToolbar={linkToolbar}
           slashMenu={slashMenu}
           sideMenu={sideMenu}
-          imageToolbar={imageToolbar}
+          fileToolbar={fileToolbar}
```
Comment (YousefED) [inline packages/react/src/editor/BlockNoteView.tsx:113]: rename to filePanel?

File: packages/react/src/editor/styles.css [hunk 9]
```diff
@@ -104,6 +104,13 @@
   font-family: var(--bn-font-family);
 }
 
+/* Custom block node view wrapper styling */
```
Comment (YousefED) [inline packages/react/src/editor/styles.css:107]: why is this changed?

**Review by YousefED — COMMENTED** [review]
added minor discussion points

File: packages/react/src/editor/styles.css (hunk 9 above)
Comment (matthewlipski) [reply packages/react/src/editor/styles.css:107]: This was to do with the blue outline for selected blocks. Previously wasn't working properly for React custom blocks, so this CSS is a fix for that.
